driver/p11_driver.mli

```diff
@@ -4,8 +4,11 @@ open P11
 
 exception CKR of RV.t
 
-module type S =
-sig
+module type S = sig
+  (** High-level interface for PKCS#11 bindings. Contains all functions in the PKCS#11
+      specification as well as helper functions to make working with PKCS#11 easier. All functions
+      take ocaml types (rather than CK_* types), and structure allocation and populate is
+      handled automatically. *)
```

Comment:
I think it's more common to put the docstring outside of the function, functor or signature it describes.  It can be just before or just after, but in this module, "just before" seems to be more common.  Same for other docstrings.

Also, I find the term "OCaml type" a bit confusing here, since everything is an OCaml type, strictly speaking.  What about core (higher-level) types?  `P11` is the core module and `Pkcs11` are the bindings that use the types from `P11`.